osidb/api_views.py

```diff
@@ -155,45 +155,53 @@ def get_valid_http_methods(cls: ViewSet, excluded: list[str] = None) -> list[str
+        "can be used to filter on related model fields. "
+        "Example: `exclude_fields=field,related_model_field.field`"
+    ),
+)
+
+include_meta_attr = OpenApiParameter(
+    "include_meta_attr",
+    type={"type": "array", "items": {"type": "string"}},
+    location=OpenApiParameter.QUERY,
+    description=(
+        "Specify which keys from meta_attr field should be retrieved, "
+        "multiple values may be separated by commas. "
+        "Dot notation can be used to specify meta_attr keys on related models. "
+        "Example: `include_meta_attr=key,related_model.key`"
+        "Use wildcards eg. `include_meta_attr=*,related_model.*` "
+        "for retrieving all the keys from meta_attr. "
+        "Omit this parameter to not include meta_attr fields at all. "
+    ),
+)
+
+
 @extend_schema_view(
     list=extend_schema(
         parameters=[
-            OpenApiParameter(
-                "include_fields",
-                type={"type": "array", "items": {"type": "string"}},
-                location=OpenApiParameter.QUERY,
-                description=(
-                    "Include only specified fields in the response. "
-                    "Multiple values may be separated by commas. Dot notation "
-                    "can be used to filter on related model fields. "
-                    "Example: `include_fields=uuid,affects.uuid,affects.trackers.uuid`"
-                ),
-            ),
-            OpenApiParameter(
-                "exclude_fields",
-                type={"type": "array", "items": {"type": "string"}},
-                location=OpenApiParameter.QUERY,
-                description=(
-                    "Exclude specified fields from the response. "
-                    "Multiple values may be separated by commas. Dot notation "
-                    "can be used to filter on related model fields. "
-                    "Example: `exclude_fields=uuid,affects.uuid,affects.trackers.uuid`"
-                ),
-            ),
-            OpenApiParameter(
-                "include_meta_attr",
-                type={"type": "array", "items": {"type": "string"}},
-                location=OpenApiParameter.QUERY,
-                description=(
-                    "Specify which keys from meta_attr field should be retrieved, "
-                    "multiple values may be separated by commas. "
-                    "Dot notation can be used to specify meta_attr keys on related models. "
-                    "Example: `include_meta_attr=bz_id,affects.ps_module,affects.trackers.bz_id`"
-                    "Use wildcards eg. `include_meta_attr=*,affects.*,affects.trackers.*` "
-                    "for retrieving all the keys from meta_attr. "
-                    "Omit this parameter to not include meta_attr fields at all. "
-                ),
-            ),
+            include_fields_param,
+            exclude_fields_param,
+            include_meta_attr,
```

Comment:
should be done in the retrieve section as well

Reply:
Unfortunately you cannot do this from a serializer point of view. `extend_schema` and  `extend_schema_view` works only on api views and `extend_schema_serializer` has limited functionality for deprecating fields, examples, etc.

We can create mixin, eg:

```python
@extend_schema_view(
    list=extend_schema(
        parameters=[include_fields_param, exclude_fields_param, include_meta_attr],
    ),
    retrieve=extend_schema(
        parameters=[include_fields_param, exclude_fields_param, include_meta_attr],
    ),
)
class ParamsModelViewSet(ModelViewSet):
    pass
```

but in some cases we also need some additional params as well and purpose of this would be defeated.

Reply:
Well apparently these `extend_schema_view` decorators can be stacked as they don't overwrite themselves, so something like this is possible:

```python
@extend_schema_view(
    list=extend_schema(
        parameters=[include_fields_param, exclude_fields_param],
    ),
    retrieve=extend_schema(
        parameters=[include_fields_param, exclude_fields_param],
    ),
)
class ParamsModelViewSet(ModelViewSet):
    pass


@extend_schema_view(
    list=extend_schema(
        parameters=[include_meta_attr],
    ),
    retrieve=extend_schema(
        parameters=[include_meta_attr],
    ),
)
class AffectView(ParamsModelViewSet):
  ...
```